internal/services/blackwhitelist_test.go

```diff
@@ -86,151 +84,205 @@ func TestGenerateProjects(t *testing.T) {
 			Project:     "native",
 			Environment: "development",
 		},
+		{
+			Project:     "native",
+			Environment: "integration",
+		},
+		{
+			Project:     "native",
+			Environment: "production",
+		},
 	}
 
+
+
 	//WHITELIST
-	t.Run("with_empty_whitelist", func(t *testing.T) {
+	t.Run("whitelist_enabled_and_projects_in_blacklist", func(t *testing.T) {
 
 		blackWhitelistData := map[string]string{
 			"blacklist": "native-development,native-integration",
 			"whitelist": "",
 		}
 
-		blackWhitelist := services.MakeBlackWhitelist(blackWhitelistData)
+		expected := []*types.Project{
```

Comment:
does this make sense? If whitelist is enabled and nothing is in whitelist, shouldn't there be nothing in the expected Project list? You don't seem to test it.